src/ui/public/Vis/AggConfig.js

```diff
@@ -259,9 +259,14 @@ define(function (require) {
     };
 
     AggConfig.prototype.makeLabel = function () {
-      if (!this.type) return '';
-      var pre = (_.get(this.vis, 'params.mode') === 'percentage') ? 'Percentage of ' : '';
-      return pre += this.type.makeLabel(this);
+      var label = (_.get(this.vis, 'params.mode') === 'percentage') ? 'Percentage of ' : '';
+      label += this.type.makeLabel(this);
+
+      if (this.params.customLabel && this.params.customLabel !== '') {
```

Comment:
This conditional is redundant: in the event that `this.params.customLabel` is truthy, it cannot also be an empty string.